### ISSUE TITLE
README.md for Compose Multiplatform/iOS development is required.

### ISSUE THREAD

**Issue by Corvus400** [DroidKaigi/conference-app-2024#789]
A fair amount of knowledge should be required, but nowhere is it mentioned what operations are needed to build the environment or in everyday development.
It might be nice if the README.md was placed in a module such as app-ios-shared and linked from the README.md in the project root.

**Comment by Corvus400** [comment]
I may be able to create a draft for now, but I'm not quite sure if it will be correct.
I would love to get some help from iOS people or people familiar with XCFramework. 🙇

**Comment by takahirom** [comment]
We are experiencing some issues with X64 architecture. However, for ARM users, we can refer to [the README in the app-ios](https://github.com/DroidKaigi/conference-app-2024/tree/main/app-ios) directory of the DroidKaigi 2024 conference app repository. I believe we are facing two distinct problems: running the app on a physical device and running it on an X64

**Comment by takahirom** [comment]
Related discussion:
https://github.com/DroidKaigi/conference-app-2024/discussions/768

**Comment by takahirom** [comment]
I've added the Trouble shooting
https://github.com/DroidKaigi/conference-app-2024/tree/main/app-ios#case3-you-are-using-x86_64-mac-and-want-to-build-the-project

**Comment by mannodermaus** [comment]
I can confirm that the build was successful on x64 after following the revised instructions! The current script will always build both debug _and_ release builds, however, which adds a lot of time to each build. Maybe we could constrain it to just the debug build for normal development, and then create a separate release pipeline for that variant?

**Comment by takahirom** [comment]
@mannodermaus I agree. Could you rename it to "x86_64Debug" or something similar, and filter the debug version as "arm64SimulatorDebug"?

**Comment by Corvus400** [comment]
@takahirom 
Thank you for your response! 🙏 
However, I can build the iPhone Simulator, but I can't seem to build on the actual iPhone device if I follow the instructions. 😭
I tried adding the command, but still no luck.

```
./gradlew iosArm64AggregateResources --no-configuration-cache
```

![スクリーンショット 2024-08-26 2 33 00](https://github.com/user-attachments/assets/88f543e3-8441-440a-ac64-5311c9e72e90)

**Comment by Corvus400** [comment]
@takahirom 
I have tried everything since then, but the build to the actual device is still not successful. 😭
We will proceed with development only in the simulator. 🫡

**Comment by mannodermaus** [comment]
@takahirom I have patched it locally so that the configuration in Xcode dictates which XCFramework to build (debug or release). This works fairly well. I'll submit this proposal as a PR shortly and you can decide what to do with it. 🙇‍♂️ 

`build_xcframework_with_xcode_environment_variable.sh`:

```diff

 # Initialize the target architecture
 target_arch=""
+target_configuration=""

+if [[ "$CONFIGURATION" == *"Debug"** ]]; then
+  target_configuration="Debug"
+elif [[ "$CONFIGURATION" == *"Release"** ]]; then
+  target_configuration="Release"
+fi

-./gradlew assembleSharedXCFramework --no-configuration-cache -Papp.ios.shared.arch=$target_arch
+./gradlew assembleShared${target_configuration}XCFramework --no-configuration-cache -Papp.ios.shared.arch=$target_arch
```

**Comment by takahirom** [comment]
@Corvus400 
I wasn't able to build as I'm seeing the error "Signing for 'DroidKaigi2024App' requires a development team. Select a development team in the Signing & Capabilities editor."

However, I think you can run `Clean Build Folder` after executing `./gradlew app-ios-shared:assembleSharedXCFramework -Papp.ios.shared.arch=arm64 --no-configuration-cache`. (In my environment, this was able to fix the "no library for this platform" error.)

@mannodermaus Thanks!

**Comment by Corvus400** [comment]
@takahirom 
Thank you very much!
I will try again when I have more time!

**Comment by Corvus400** [comment]
@takahirom 
I was able to install it on the actual iPhone device!

The steps we tried are as follows

1. run Clean Project in Android Studio
2. . Run /gradlew app-ios-shared:assembleSharedXCFramework -Papp.ios.shared.arch=arm64 --no-configuration-cache
3. run make bootstrap
4. try to install on the actual device
5. build error
6. run Clean Build Folder in Xcode
7. run . Execute /gradlew app-ios-shared:assembleSharedXCFramework -Papp.ios.shared.arch=arm64 --no-configuration-cache
8. try to install on the actual device
9. Installation succeeded.
